RPLCD/i2c.py

```diff
@@ -57,18 +82,26 @@ def __init__(self, address, port=1,
         :type auto_linebreaks: bool
         :param backlight_enabled: Whether the backlight is enabled initially. Default: True.
         :type backlight_enabled: bool
-
+        :param i2c_expander: Set your i2c chip type. "PCF8574" or "MCP23008" supported. Default: "PCF8574"
         """
+
         # Set own address and port.
         self.address = address
         self.port = port
 
+        # Set i2c expander, "PCF8574" and "MCP23008" are supported.
+        self.i2c_expander = i2c_expander
+
         # Currently the I2C mode only supports 4 bit communication
         self.data_bus_mode = c.LCD_4BITMODE
 
         # Set backlight status
-        self._backlight = c.LCD_BACKLIGHT if backlight_enabled else c.LCD_NOBACKLIGHT
-
+        if(self.i2c_expander == "PCF8574"):
+            self._backlight = c.LCD_BACKLIGHT if backlight_enabled else c.LCD_NOBACKLIGHT
+        elif(self.i2c_expander == "MCP23008"):
+            self._backlight = c.MCP23008_BACKLIGHT if backlight_enabled else c.MCP23008_NOBACKLIGHT
+        else:
+            raise NotImplementedError('I2C expander is not supported.')
```

Comment:
Absolutely.  But I'm not sure all of them are needed. Probably better to do it once when initiated.
And will rename i2c_expander to _i2c_expander since it's a really bad idea to let users change it after the class is initiated.